src/about-this-guide.md

```diff
@@ -102,7 +101,7 @@ You might also find the following sites useful:
 [Miri]: https://github.com/rust-lang/miri
 [@bors]: https://github.com/bors
 [GitHub repository]: https://github.com/rust-lang/rustc-dev-guide/
-[rustc API docs]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/
+[rustc API docs]: https://doc.rust-lang.org/nightly/nightly-rustc/
```

Comment:
this does not point to the right place... it displays a page of all crates included in rust-lang/rust

Reply:
Hmm. This is a rustdoc limitation, we pass it all unstable crates at once. I can change it back to pointing directly at rustc_middle, but the sidebar will still show all the crates: 
![Uploading image.png…]()

Reply:
Hmm. This is a rustdoc limitation, we pass it all unstable crates at once. I can change it back to pointing directly at rustc_middle, but the sidebar will still show all the crates: 
![image](https://user-images.githubusercontent.com/23638587/230793549-e7b86b9f-1c6e-4572-ab8a-57e172c1ba04.png)

Reply:
what do you suggest

Reply:
I think we should link to the index page, but expand the link description a bit to make it clear this includes more crates than just rustc.

Reply:
a curious things is that my local docs build (done with `x doc library compiler --stage 1`) only shows rustc crates actually (with the exception of `coverage_test_macros`), unlike the hosted docs

Reply:
You've explicitly specified `doc library compiler`. To replicate CI you need `./configure --set compiler-docs && x doc` without any filter, which will also build docs for cargo/rustdoc/miri/rustfmt.

Reply:
heh, indeed

Reply:
One other thing is that rustc_middle is said to be the "main crate" of rustc... do you think this is still accurate, and in such a case, would it be okay to point to it as the landing page for "rustc docs"?

Reply:
I would rather not block this change on the link, I don't feel strongly about it. I'll revert the change to "About this guide".

Reply:
I agree... the discussion should happen elsewhere. Such is why I am a fan of PRs avoiding drive-by changes.